azurerm/internal/services/loganalytics/log_analytics_data_export_resource.go

```diff
@@ -0,0 +1,199 @@
+		Update: resourceArmOperationalinsightsDataExportCreateUpdate,
+		Delete: resourceArmOperationalinsightsDataExportDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     validate.LogAnalyticsDataExportName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"workspace_resource_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"destination_resource_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"table_names": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.NoZeroValues,
+				},
+			},
+
+			"export_rule_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmOperationalinsightsDataExportCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.DataExportClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	workspace, err := parse.LogAnalyticsWorkspaceID(d.Get("workspace_resource_id").(string))
+	if err != nil {
+		return fmt.Errorf("%v", err)
+	}
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, workspace.Name, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for present of existing Log Analytics Data Export Rule %q (Resource Group %q / workspaceName %q): %+v", name, resourceGroup, workspace.Name, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_log_analytics_data_export_rule", *existing.ID)
+		}
+	}
+
+	destination := &operationalinsights.Destination{
+		ResourceID: utils.String(d.Get("destination_resource_id").(string)),
+	}
+
+	parameters := operationalinsights.DataExport{
+		DataExportProperties: &operationalinsights.DataExportProperties{
+			Destination: destination,
+			TableNames:  utils.ExpandStringSlice(d.Get("table_names").(*schema.Set).List()),
+			Enable:      utils.Bool(d.Get("enabled").(bool)),
+		},
+	}
+
```

Comment:
_minor_ this should be
```suggestion
	parameters := operationalinsights.DataExport{
		DataExportProperties: &operationalinsights.DataExportProperties{
			Destination: &operationalinsights.Destination{
				ResourceID: utils.String(d.Get("destination_resource_id").(string)),
			},
			TableNames:  utils.ExpandStringSlice(d.Get("table_names").(*schema.Set).List()),
			Enable:      utils.Bool(d.Get("enabled").(bool)),
		},
	}

```